src/wallet/wallet.h

```diff
@@ -758,6 +759,9 @@ class CWallet final : public WalletStorage, public interfaces::Chain::Notificati
     /** Interface for accessing chain state. */
     interfaces::Chain* m_chain;
 
+    /** Interface for accessing CoinJoin state. */
+    const std::unique_ptr<interfaces::CoinJoin::Loader>& m_coinjoin_loader;
```

Comment:
just hold const ref to underlying

Reply:
unfortunately due to initialization order fiask, CoinjoinLoader is not created yet when this reference is initialized.
See this TODO: https://github.com/dashpay/dash/pull/5767/files#diff-6a8ef76c60f30a6ca67d9f0e478fd02989c4b7fbc4c3116f80e13d873d5775e6R170

and this: https://github.com/dashpay/dash/pull/5767/files#diff-6a8ef76c60f30a6ca67d9f0e478fd02989c4b7fbc4c3116f80e13d873d5775e6R170

I removed that many references to std::unique_ptr as possible in [this commit ](https://github.com/dashpay/dash/pull/5767/commits/2615c18b2ed2ad5ad96426c1a06c2637b50d3986) but further refactoring out-of-scope of this PR.
Same situation as with PeerManager as you met in other PR.